apps/shipments/serializers/shipment.py

```diff
@@ -184,7 +188,30 @@ def create(self, validated_data):
             if 'device' in self.context:
                 extra_args['device'] = self.context['device']
 
-            return Shipment.objects.create(**validated_data, **extra_args)
+            if 'tags' in validated_data:
+                shipment_tags = validated_data.pop('tags')
+                shipment = Shipment.objects.create(**validated_data, **extra_args)
+                for shipment_tag in shipment_tags:
+                    ShipmentTag.objects.create(
+                        tag_type=shipment_tag['tag_type'],
+                        tag_value=shipment_tag['tag_value'],
+                        shipment_id=shipment.id,
+                        owner_id=shipment.owner_id
+                    )
+            else:
+                shipment = Shipment.objects.create(**validated_data, **extra_args)
```

Comment:
Yes, but if the tags are in it we need to be sure to remove it from the validated data before trying to create, that's why it's separated. I figured it would be better to have the if/else statement than having a check before the create if `tags` in validated data and removing it then another after the shipment was created for creating the tags.

Reply:
I think something along the lines of the following would improve readability here:
```
shipment_tags = validated_data.pop('tags', [])
shipment = Shipment.objects.create(**validated_data, **extra_args)
for shipment_tag in shipment_tags:
    ShipmentTag.objects.create(
        tag_type=shipment_tag['tag_type'],
        tag_value=shipment_tag['tag_value'],
        shipment_id=shipment.id,
        owner_id=shipment.owner_id
    )
```